### DISCUSSION TITLE
Welcome to paru Discussions!

### DISCUSSION THREAD

**Discussion by htdebeer** [htdebeer/paru#68] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by iandol** [comment]
Hi all, I use paru to write some filters for a workflow that automates compiling documents from the macOS writer's app Scrivener to Pandoc outputs for academic work. Compared to Lua, Ruby has a nicer syntax and debugging tools, so it makes using a ruby backend like paru really easy. Thanks to Huub for making such a great pandoc interface and also the "best" pandoc automator, pandocomatic!

**Reply by htdebeer** [reply]
You're welcome!

Nowadays, I find myself (re)writing some of my filters in lua instead of paru. Although I've not measured it, it seems to make pandoc run quite a bit faster. 

But I agree with you, paru's API is nicer. In particular, I miss the following two things in lua filter:

- the `inner_markdown` method, and to a lesser degree the `outer_markdown` method. It's very convenient not to construct a tree of AST nodes manually. Instead, with paru you write `some_node.inner_markdown = "**#{some_node.inner_markdown}**"`.
- replacing inline nodes by blocks and vice-versa. In lua filter you have to replace inline by inline and block by block. I don't use it often, but it is nice to have.

**Comment by lygaret** [comment]
Hey y'all 👋 

I'm Jon Raphaelson, a ruby developer in Boulder County, Colorado - working with `paru` as part of a markdown-based note taking application, and realizing how cool pandoc is in the process :)

Thanks @htdebeer for the great library and introducing me to what pandoc actually is!
